data_store/transformation/pathfinders/pf_transform_r2.py

```diff
@@ -0,0 +1,662 @@
+from datetime import datetime
+
+import pandas as pd
+
+from data_store.const import (
+    PF_REPORTING_ROUND_NUMBER_TO_OBSERVATION_DATES,
+    PF_REPORTING_ROUND_NUMBER_TO_SUBMISSION_DATES,
+    FundTypeIdEnum,
+)
+from data_store.transformation.utils import create_dataframe, extract_postcodes
+
+FAS_REPORTING_PERIOD_HEADERS_TO_DATES = {
+    "Total cumulative actuals to date, (Up to and including Mar 2024)": {
+        "start": None,
+        "end": datetime(2024, 3, 31, 23, 59, 59),
+    },
+    "Financial year 2024 to 2025, (Apr to Jun)": {
+        "start": datetime(2024, 4, 1),
+        "end": datetime(2024, 6, 30, 23, 59, 59),
+    },
+    "Financial year 2024 to 2025, (Jul to Sep)": {
+        "start": datetime(2024, 7, 1),
+        "end": datetime(2024, 9, 30, 23, 59, 59),
+    },
+    "Financial year 2024 to 2025, (Oct to Dec)": {
+        "start": datetime(2024, 10, 1),
+        "end": datetime(2024, 12, 31, 23, 59, 59),
+    },
+    "Financial year 2024 to 2025, (Jan to Mar)": {
+        "start": datetime(2025, 1, 1),
+        "end": datetime(2025, 3, 31, 23, 59, 59),
+    },
+    "Financial year 2025 to 2026, (Apr to Jun)": {
+        "start": datetime(2025, 4, 1),
+        "end": datetime(2025, 6, 30, 23, 59, 59),
+    },
+    "Financial year 2025 to 2026, (Jul to Sep)": {
+        "start": datetime(2025, 7, 1),
+        "end": datetime(2025, 9, 30, 23, 59, 59),
+    },
+    "Financial year 2025 to 2026, (Oct to Dec)": {
+        "start": datetime(2025, 10, 1),
+        "end": datetime(2025, 12, 31, 23, 59, 59),
+    },
+    "Financial year 2025 to 2026, (Jan to Mar)": {
+        "start": datetime(2026, 1, 1),
+        "end": datetime(2026, 3, 31, 23, 59, 59),
+    },
+}
+
+
+OUTPUT_REPORTING_PERIOD_HEADERS_TO_DATES = {
+    "Total cumulative outputs to date, (Up to and including Mar 2024)": {
+        "start": None,
```

Comment:
This seems to be causing errors because `output_data.start_date` is non-nullable. Is this expected?

<img width="1609" alt="image" src="https://github.com/user-attachments/assets/4f60e371-3971-4ff0-9401-eb307344864c">